silnlp/nmt/translate.py

```diff
@@ -49,86 +140,39 @@ def main() -> None:
-    book: Optional[str] = args.book
-    step_str = "avg" if step == -1 else str(step)
-    if book is not None:
-        src_project: Optional[str] = args.src_project
-        if src_project is None:
-            if len(config.src_projects) != 1:
-                raise RuntimeError("A source project must be specified.")
-            src_project = next(iter(config.src_projects))
-
-        default_output_dir = config.exp_dir / "infer" / step_str
-        output_path: Optional[Path] = None if args.output_usfm is None else Path(args.output_usfm)
-        if output_path is None:
-            book_num = book_id_to_number(book)
-            output_path = default_output_dir / f"{book_file_name_digits(book_num)}{book}.SFM"
-        elif output_path.name == output_path:
-            output_path = default_output_dir / output_path
+    translator = TranslationTask(
+        name=args.experiment,
+        checkpoint=args.checkpoint,
+        memory_growth=args.memory_growth,
+        clearml_queue=args.clearml_queue,
+    )
 
-        output_path.parent.mkdir(exist_ok=True, parents=True)
-        translator.translate_book(src_project, book, output_path, trg_iso=trg_iso)
+    if args.book is not None:
+        translator.translate_book(
+            args.book, src_project=args.src_project, output_usfm=args.output_usfm, trg_iso=args.trg_iso
+        )
     elif args.src_prefix is not None:
-        if args.trg_prefix is None:
-            raise RuntimeError("A target file prefix must be specified.")
-        if args.start_seq is None or args.end_seq is None:
-            raise RuntimeError("Start and end sequence numbers must be specified.")
-
-        checkpoint_name = "averaged checkpoint" if step == -1 else f"checkpoint {step}"
-        print(f"Translating using {checkpoint_name}...")
-        cwd = Path.cwd()
-        for i in range(args.start_seq, args.end_seq + 1):
-            file_num = f"{i:04d}"
-            src_file_path = cwd / f"{args.src_prefix}{file_num}.txt"
-            trg_file_path = cwd / f"{args.trg_prefix}{file_num}.txt"
-            if src_file_path.is_file() and not trg_file_path.is_file():
-                start = time.time()
-                translator.translate_text_file(src_file_path, trg_file_path, trg_iso=trg_iso)
-                end = time.time()
-                print(f"Translated {src_file_path.name} to {trg_file_path.name} in {((end-start)/60):.2f} minutes")
+        translator.translate_parts(args.src_prefix, args.trg_prefix, args.start_seq, args.end_seq, args.trg_iso)
     elif args.src is not None:
-        src_file_path = Path(args.src)
-        default_output_dir = config.exp_dir / "infer" / step_str
-        trg_file_path = default_output_dir / src_file_path.name if args.trg is None else Path(args.trg)
-        trg_file_path.parent.mkdir(exist_ok=True, parents=True)
-        translator.translate_text_file(src_file_path, trg_file_path, trg_iso=trg_iso)
+        translator.translate_src(args.src, args.trg_iso, args.trg)
```

Comment:
I think you missed updating this method name.

Reply:
Correct.  Fixed.